src/nxt_sprintf.c

```diff
@@ -554,6 +555,11 @@ nxt_vsprintf(u_char *buf, u_char *end, const char *fmt, va_list args)
 
     copy:
 
+        if (nxt_slow_path(p == NULL)) {
+            p = (u_char *) null;
+            length = nxt_length(null);
+        }
+
         buf = nxt_cpymem(buf, p, nxt_min((size_t) (end - buf), length));
```

Comment:
`nxt_min((size_t) (end - buf), length)`

might be a problem since we don't control `end` and `buf` when `p = NULL`.

I suggest:

```c
if (nxt_slow_path(p == NULL)) {
    p = null;
    length = nxt_length(null);

} else {
    length = nxt_min((size_t) (end - buf), length);
}

buf = nxt_cpymem(buf, p, length);
continue;
```

Reply:
Also, thank you for noticing, fixed.